narwhal/worker/src/worker.rs

```diff
@@ -453,22 +459,17 @@ struct TxReceiverHandler<V> {
 }
 
 impl<V: TransactionValidator> TxReceiverHandler<V> {
-    async fn wait_for_shutdown(mut rx_reconfigure: watch::Receiver<ReconfigureNotification>) {
-        loop {
-            let result = rx_reconfigure.changed().await;
-            result.expect("Committee channel dropped");
-            let message = rx_reconfigure.borrow().clone();
-            if let ReconfigureNotification::Shutdown = message {
-                break;
-            }
+    async fn wait_for_shutdown(mut tx_shutdown: ConditionalBroadcastReceiver) {
+        match tx_shutdown.receiver.recv().await {
+            Ok(()) | Err(_) => (),
```

Comment:
Since we don't use the result of the receiver you could alternatively do the following:
```
_ = tx_shutdown.receiver.recv().await;
```